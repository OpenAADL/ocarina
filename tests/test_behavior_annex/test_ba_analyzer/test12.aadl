
package exemple
public
   with Base_Types ;

thread th
end th;


thread implementation th.good

annex behavior_specification {**

states
 uniqueState : initial complete final state ;

transitions
 uniqueState -[on dispatch]-> uniqueState ;  
  
**};

end th.good;

thread implementation th.error
annex behavior_specification {**

states
 compState : initial complete state ;
 exeState : state ;
 finalState : final state ; 

transitions
-- Errors: 
--          ** transitions out of complete states must have dispatch conditions.
  compState -[]-> finalState ;
 
**};
end th.error;

end exemple ;
