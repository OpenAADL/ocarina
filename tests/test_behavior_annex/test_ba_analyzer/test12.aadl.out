test12.aadl:35:03: "compState" Transitions out of complete states must have dispatch condition
Cannot analyze BEHAVIOR_SPECIFICATION specifications
